executor/builder.go

```diff
@@ -3765,6 +3765,21 @@ func (b *executorBuilder) buildIndexReader(v *plannercore.PhysicalIndexReader) E
 	}
 
 	if is.Index.Global {
+		tmp, ok := b.is.TableByID(ret.table.Meta().ID)
+		if !ok {
+			b.err = err
+			return nil
+		}
+		tbl, ok := tmp.(table.PartitionedTable)
+		if !ok {
+			b.err = exeerrors.ErrBuildExecutor
+			return nil
+		}
```

Comment:
maybe could remove those codes and take L3786 - L3787 to L3766

Reply:
@mjonss told me I should check explicitly before. You can talk about it and I will follow your common advice.

Reply:
I handle the err (ErrTableNotExists) and still check the type transform explicitly here. @Defined2014